src/Forms/Checkbox.js

```diff
@@ -40,8 +42,8 @@ const Checkbox = React.forwardRef(({ checked, className, defaultChecked, disable
                 disabled={disabled}>
                 <input
                     {...inputProps}
-                    aria-checked={getCheckStatus(checked, indeterminate)}
-                    checked={checked || defaultChecked}
+                    aria-checked={getCheckStatus(isChecked, indeterminate)}
+                    checked={isChecked}
```

Comment:
Do the prop descriptions for `checked` and `defaultChecked` need to be adjusted?